jasmin_manage/models/service.py

```diff
@@ -57,6 +59,22 @@ def get_event_aggregates(self):
     def natural_key(self):
         return self.category.name, self.name
 
+    def get_num_active_requirements(self):
+        # Checks whether there are any active reqs. if so returns True, else False
+        return self.requirements.filter(status=50).count()
+
+    def get_parent_consortium(self):
```

Comment:
I think this and the following 3 could be decorated with `@django.urils.functional.cached_property` for speed: https://docs.djangoproject.com/en/5.1/topics/db/managers/#adding-extra-manager-methods

I think you'd have to move them to the Model rather than the manager for this to work right though(?)

You'd them have to access them as a property instead of a method() below.

Reply:
I've just removed the methods from the model and changed the methods in the serializer?
```
def get_consortium(self, obj):
        # Get the parent project's consortium
        return obj.project.consortium.id

    def get_consortium_fairshare(self, obj):
        # Get the fairshare for the parent consortium
        return obj.project.consortium.fairshare

    def get_project_fairshare(self, obj):
        # Get the fairshare for the parent consortium
        return obj.project.fairshare
```

Because I'm just accessing a property anyway?

Reply:
Oh yeah that works too